packages/web/src/components/track/PremiumConditionsPill.tsx

```diff
@@ -1,28 +1,51 @@
+import { useCallback, MouseEvent } from 'react'
+
 import {
   formatPrice,
   isPremiumContentUSDCPurchaseGated,
-  PremiumConditions
+  PremiumConditions,
+  premiumContentActions
 } from '@audius/common'
-import { IconLock } from '@audius/stems'
-import cn from 'classnames'
+import { HarmonyButton, HarmonyButtonSize, IconLock } from '@audius/stems'
+import { useDispatch } from 'react-redux'
 
+import { useModalState } from 'common/hooks/useModalState'
 import LoadingSpinner from 'components/loading-spinner/LoadingSpinner'
 
 import styles from './PremiumConditionsPill.module.css'
 
+const { setPurchaseContentId } = premiumContentActions
+
 const messages = {
   unlocking: 'Unlocking',
   locked: 'Locked'
 }
 
+const usePurchaseModal = () => {
+  const dispatch = useDispatch()
+  const [, setPurchaseModalVisibility] = useModalState('PremiumContentPurchase')
```

Comment:
I feel like since we have createModal now and it has the ability to pass the state through in the `show` call, we should consider migrating this modal over to that pattern.

Reply:
migrated!